src/scene/procedural.js

```diff
@@ -232,6 +232,10 @@ function calculateTangents(positions, normals, uvs, indices) {
  * @param {number[]} [opts.blendWeights] - An array of 4-dimensional bone weights where each
  * component is in the range 0 to 1 and the sum of the weights should equal 1.
  * @param {number[]} [opts.indices] - An array of triangle indices.
+ * @param {boolean} [opts.storageVertex] - Defines if the vertex buffer of the mesh can be used as
+ * a storage buffer by a compute shader. Defaults to false. Only supported on WebGPU.
+ * @param {boolean} [opts.storageIndex] - Defines if the index buffer of the mesh can be used as
+ * a storage buffer by a compute shader. Defaults to false. Only supported on WebGPU.
```

Comment:
I'm not sure why I feel uneasy about adding new WebGPU-only options to this API. I almost feel like shape creation should be very simple and pure. Like maybe it should return just bare position, normal, uv and index arrays. And you create the mesh yourself in whatever form you like (e.g. whether you want it to be used by a compute shader etc).